internal/rpc/starknet/types_test.go

```diff
@@ -0,0 +1,276 @@
+		t.Run(tt.name, func(t *testing.T) {
+			var got StorageKey
+			if err := got.UnmarshalJSON(tt.data); (err != nil) != tt.err {
+				t.Errorf("StorageKey.UnmarshalJSON(), unexpected error: %v", err)
+			}
+			assert.DeepEqual(t, got, tt.want)
+		})
+	}
+}
+
+func TestStorageKeyToFelt(t *testing.T) {
+	tests := []struct {
+		name       string
+		storageKey StorageKey
+		want       *felt.Felt
+	}{
+		{
+			name:       "test 1",
+			storageKey: "0x07352d792298a2578d6ef20e80bce32473fff67b9b12b1bc431982287190291a",
+			want:       new(felt.Felt).SetHex("0x07352d792298a2578d6ef20e80bce32473fff67b9b12b1bc431982287190291a"),
+		},
+		{
+			name:       "test 2",
+			storageKey: "0x00898cca7dbf84c2213f3a00e84775013bc991bc104d3a00952ce4bc166a4a1a",
+			want:       new(felt.Felt).SetHex("0x00898cca7dbf84c2213f3a00e84775013bc991bc104d3a00952ce4bc166a4a1a"),
+		},
+		{
+			name:       "test 2",
+			storageKey: "0x0000000000000000000000000000000000000000000000000000000000000005",
+			want:       new(felt.Felt).SetHex("0x0000000000000000000000000000000000000000000000000000000000000005"),
+		},
+	}
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			got := tt.storageKey.Felt()
+			if !got.Equal(tt.want) {
+				t.Fatalf("StorageKey.Felt() = %v, want %v", got, tt.want)
+			}
+		})
+	}
+}
+
+func TestRpcFelt_UnmarshalJSON(t *testing.T) {
+	tests := []struct {
+		name string
+		data []byte
+		want RpcFelt
+		err  bool
+	}{
+		{
+			name: "empty data",
+			data: []byte{},
+			want: "",
+			err:  true,
+		},
+		{
+			name: "invalid felt",
+			data: []byte(`"0xad328a71faf48c5c3857e99f20a77b18522480956d1cd5bff1ff2df3c8b427b"`),
+			want: "",
+			err:  true,
+		},
+		{
+			name: "valid felt",
+			data: []byte(`"0x02c2bb91714f8448ed814bdac274ab6fcdbafc22d835f9e847e5bee8c2e5444e"`),
+			want: "0x02c2bb91714f8448ed814bdac274ab6fcdbafc22d835f9e847e5bee8c2e5444e",
+			err:  false,
+		},
+	}
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			var got RpcFelt
+			if err := got.UnmarshalJSON(tt.data); (err != nil) != tt.err {
+				t.Errorf("RpcFelt.UnmarshalJSON(), unexpected error: %v", err)
+			}
+			assert.DeepEqual(t, got, tt.want)
+		})
+	}
+}
+
+func TestRpcFeltToFelt(t *testing.T) {
+	tests := []struct {
+		name    string
+		rpcFelt RpcFelt
+		want    *felt.Felt
+	}{
+		{
+			name:    "test 1",
+			rpcFelt: "0x045c61314be4da85f0e13df53d18062e002c04803218f08061e4b274d4b38537",
+			want:    new(felt.Felt).SetHex("0x045c61314be4da85f0e13df53d18062e002c04803218f08061e4b274d4b38537"),
+		},
+		{
+			name:    "test 2",
+			rpcFelt: "0x0320e37cf7c972458a3edf08ab51f2ab7596857706af174a3d5be4e46f16c63e",
+			want:    new(felt.Felt).SetHex("0x0320e37cf7c972458a3edf08ab51f2ab7596857706af174a3d5be4e46f16c63e"),
+		},
+		{
+			name:    "test 2",
+			rpcFelt: "0x048636a22c6c74f5631b00c66ac3c8ab4714aa308ddd214af4089ccdfcee0f81",
+			want:    new(felt.Felt).SetHex("0x048636a22c6c74f5631b00c66ac3c8ab4714aa308ddd214af4089ccdfcee0f81"),
+		},
```

Comment:
Same as above. If you just want to enumerate multiple felts, then change your test struct to the following 

```go
tests []struct{
  rpcFelt RpcFelt
  want *felt.Felt
}
```

Then iterate over the tests and assert on each case, similar to [this](https://github.com/NethermindEth/juno/blob/62f5d5d26001e1b60e8f2957f21e30282d22ce4e/internal/utils/datadir_test.go#L78).

Reply:
Done -> https://github.com/NethermindEth/juno/pull/377/commits/58b626cb47457c9890f4765a4a8288ea82bea185